src/main/webapp/app/pages/curation/CurationPage.tsx

```diff
@@ -48,6 +50,29 @@ export const CurationPage = (props: ICurationPageProps) => {
   const firebaseMetaGeneReviewPath = `${getFirebaseMetaGenePath(isGermline, hugoSymbol)}/review`;
   const firebaseMetaCurrentReviewerPath = `${firebaseMetaGeneReviewPath}/currentReviewer`;
 
+  useEffect(() => {
+    async function checkIfGeneExists() {
+      if (props.firebaseDb && hugoSymbol) {
+        const snapshot = await get(ref(props.firebaseDb, firebaseGenePath));
+        if (!snapshot.exists()) {
+          try {
+            await props.createGene(hugoSymbol, isGermline);
+          } catch (error) {
+            notifyError(error);
+            if (isGermline) {
+              history.push(PAGE_ROUTE.CURATION_GERMLINE);
+            } else {
+              history.push(PAGE_ROUTE.CURATION_SOMATIC);
+            }
+          }
+        }
+        setFirebaseGeneExists(true);
+      }
+    }
+
+    checkIfGeneExists();
+  }, [firebaseGenePath, setFirebaseGeneExists, props.firebaseDb, hugoSymbol, isGermline]);
```

Comment:
setFirebaseGeneExists as part of dependency?

Reply:
It doesn't actually affect anything (that reference never changes), but it considered best practice to put all reactive values (props, variables, and functions) used by useEffect in the dependency array: https://react.dev/reference/react/useEffect#specifying-reactive-dependencies.